### ISSUE TITLE
beforeAll() doesn't cover __construct() 

### ISSUE THREAD

**Issue by samsonasik** [kahlan/kahlan#220]
I'm not sure if it is intended behaviour, I'm just seeing it not covered with `beforeAll()`

```php
describe('Foo', function () {

    beforeAll(function () {

        $this->model = new Foo();
        
    });

    describe('->method()', function () {

        // ..
        
    });
    
});
```

It shows as red in coverage. 
![screen shot 2017-02-02 at 8 24 11 am](https://cloud.githubusercontent.com/assets/459648/22533571/68c43f64-e921-11e6-870f-9bf3fe3c191e.png)

While when switch to `beforeEach()`, it covered.

**Comment by jails** [comment]
At the very beginning only code executed inside `it()` was considered as covered. Then I considered that `beforeEach()/afterEach()` was just a feature to avoid some copy/paste so I changed the reporting to also consider code executed in `beforeEach()/afterEach()` to be covered.

However `beforeAll()/afterAll()` can't be attached to any `it()`. So I prefered to not considering them as part of the spec suite but more as part of some initialization step.
